Modules/_multiprocessing/multiprocessing.c

```diff
@@ -186,35 +198,38 @@ static PyMethodDef module_methods[] = {
 static int
 multiprocessing_exec(PyObject *module)
 {
+    _multiprocessingstate *state = get_module_state(module);
 #ifdef HAVE_MP_SEMAPHORE
 
-    /* Add _PyMp_SemLock type to module */
-    if (PyModule_AddType(module, &_PyMp_SemLockType) < 0) {
+    state->semlock_type = (PyTypeObject *)PyType_FromModuleAndSpec(
+                module, &_PyMp_SemLockType_spec, NULL);
```

Comment:
The module state is unnecessary here.
With this change:
```suggestion
    PyObject *semlock_type = (PyTypeObject *)PyType_FromModuleAndSpec(
                module, &_PyMp_SemLockType_spec, NULL);
```
and adding some `Py_DECREF(semlock_type)`, you can change `state->semlock_type` to just `semlock_type` in this function, and remove `_multiprocessingstate`, `get_module_state`, `multiprocessing_traverse`, `multiprocessing_clear`, `multiprocessing_dealloc`.

Reply:
removed